src/authentication/team-token.ts

```diff
@@ -0,0 +1,88 @@
+import * as Knex from 'knex';
+import * as moment from 'moment';
+
+const randomstring = require('randomstring');
+
+export const teamTokenLength = 16;
+
+export interface TeamToken {
+  teamId: number;
+  token: string;
+  createdAt: number | moment.Moment;
+}
+
+/**
+ * Creates an sql query for fetching valid, i.e. latest, team tokens.
+ * The results can be filtered by specifying a token,
+ * a teamId or both.
+ */
+export function teamTokenQuery(db: Knex, token?: string, teamId?: number) {
+  const latestTokens = db('teamtoken')
+    .select('teamId')
+    .max('createdAt AS latestStamp')
+    .groupBy('teamId')
+    .as('latest');
+  const query = db('teamtoken')
+    .join(latestTokens, ((join: any) => join
+      .on('teamtoken.teamId', '=', 'latest.teamId')
+      .andOn('teamtoken.createdAt', '=', 'latest.latestStamp')) as any,
+    );
+  if (token) {
+    if (token.length !== teamTokenLength || !token.match(/^\w+$/)) {
```

Comment:
Ok. But the main questions still stand: what are we benefitting with this check? Why is it being done? What does it matter if we try to search for a string that's a different length or of a different format?

I completely understand e.g. checking the length and format when we want to store a token to make sure we don't put wrong stuff into our database, but I simply don't understand the validation here. Unless there's some security implication that I haven't thought about, to me, this just seems to add unnecessary complexity to our future code and adds one potential source of problems.